Index: cairo/src/cairo-mutex.c
===================================================================
--- cairo.orig/src/cairo-mutex.c	2010-06-26 16:10:44.672182790 +0200
+++ cairo/src/cairo-mutex.c	2010-06-26 16:10:58.177182650 +0200
@@ -35,7 +35,7 @@
 
 #include "cairo-mutex-private.h"
 
-#define CAIRO_MUTEX_DECLARE(mutex) cairo_mutex_t mutex = CAIRO_MUTEX_NIL_INITIALIZER;
+#define CAIRO_MUTEX_DECLARE(mutex) cairo_private cairo_mutex_t mutex = CAIRO_MUTEX_NIL_INITIALIZER;
 #include "cairo-mutex-list-private.h"
 #undef   CAIRO_MUTEX_DECLARE
 
