--- cairo-1.0.2/configure	2005-10-04 01:48:32.000000000 +0100
+++ configure	2005-11-25 04:21:03.691181696 +0000
@@ -24985,6 +24985,10 @@
 rm -f conftest.err conftest.$ac_objext \
       conftest$ac_exeext conftest.$ac_ext
 fi
+
+echo "$as_me: Disabling FT_GlyphSlot_Embolden" >&5
+eval "$as_ac_var=no"
+
 echo "$as_me:$LINENO: result: `eval echo '${'$as_ac_var'}'`" >&5
 echo "${ECHO_T}`eval echo '${'$as_ac_var'}'`" >&6
 if test `eval echo '${'$as_ac_var'}'` = yes; then
