Index: cairo/configure.ac
===================================================================
--- cairo.orig/configure.ac	2010-12-28 14:43:02.291226995 +0100
+++ cairo/configure.ac	2010-12-28 14:43:18.191141863 +0100
@@ -10,6 +10,7 @@
 AC_CONFIG_SRCDIR(src/cairo.h)
 AC_CONFIG_HEADERS(config.h)
 AM_INIT_AUTOMAKE([1.9.6 gnu -Wall no-define])
+AM_MAINTAINER_MODE
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 AC_LIBTOOL_WIN32_DLL dnl Must be called before AC_PROG_LIBTOOL
 AC_PROG_LIBTOOL dnl ([1.4]) Don't remove!
