Index: cairo/src/Makefile.sources
===================================================================
--- cairo.orig/src/Makefile.sources	2010-06-14 09:34:16.178382562 +0200
+++ cairo/src/Makefile.sources	2010-06-14 09:42:22.494391992 +0200
@@ -282,8 +282,8 @@
 		    $(NULL)
 endif
 
-cairo_qt_headers = cairo-qt.h
-cairo_qt_sources = cairo-qt-surface.cpp
+#cairo_qt_headers = cairo-qt.h
+#cairo_qt_sources = cairo-qt-surface.cpp
 
 cairo_quartz_headers = cairo-quartz.h
 cairo_quartz_private = cairo-quartz-private.h
@@ -300,8 +300,8 @@
 
 cairo_win32_font_sources = cairo-win32-font.c
 
-cairo_skia_headers =
-cairo_skia_sources = cairo-skia-surface.cpp
+#cairo_skia_headers =
+#cairo_skia_sources = cairo-skia-surface.cpp
 
 cairo_os2_headers = cairo-os2.h
 cairo_os2_private = cairo-os2-private.h
