diff --git a/src/Makefile.sources b/src/Makefile.sources
index c1d88e6..cc296fa 100644
--- a/src/Makefile.sources
+++ b/src/Makefile.sources
@@ -281,8 +281,8 @@ cairo_xcb_sources += \
 		    $(NULL)
 endif
 
-cairo_qt_headers = cairo-qt.h
-cairo_qt_sources = cairo-qt-surface.cpp
+#cairo_qt_headers = cairo-qt.h
+#cairo_qt_sources = cairo-qt-surface.cpp
 
 cairo_quartz_headers = cairo-quartz.h
 cairo_quartz_private = cairo-quartz-private.h
@@ -299,8 +299,8 @@ cairo_win32_sources = cairo-win32-surface.c cairo-win32-printing-surface.c
 
 cairo_win32_font_sources = cairo-win32-font.c
 
-cairo_skia_headers =
-cairo_skia_sources = cairo-skia-surface.cpp
+#cairo_skia_headers =
+#cairo_skia_sources = cairo-skia-surface.cpp
 
 cairo_os2_headers = cairo-os2.h
 cairo_os2_private = cairo-os2-private.h
