Index: cairo/configure.ac
===================================================================
--- cairo.orig/configure.ac	2010-06-26 16:10:44.572181390 +0200
+++ cairo/configure.ac	2010-06-26 16:10:59.932185096 +0200
@@ -15,6 +15,8 @@
 DOLT dnl Make my libtool fast!
 GTK_DOC_CHECK([1.6])
 
+AM_MAINTAINER_MODE
+
 AC_SYS_LARGEFILE
 
 dnl ===========================================================================
